erpnext/vehicles/doctype/vehicle_booking_order/vehicle_booking_order.py

```diff
@@ -1040,3 +1054,68 @@ def update_allocation_booked(vehicle_allocation, is_booked, is_cancelled):
 	is_cancelled = cint(is_cancelled)
 	frappe.db.set_value("Vehicle Allocation", vehicle_allocation, {"is_booked": is_booked, "is_cancelled": is_cancelled},
 		notify=True)
+
+
+@frappe.whitelist()
+def send_customer_vehicle_anniversary_notifications():
+	if not automated_vehicle_anniversary_enabled():
+		return
+
+	now_dt = now_datetime()
+	date_today = getdate(now_dt)
+
+	reminder_dt = get_anniversary_scheduled_time(date_today)
+	if now_dt < reminder_dt:
+		return
+
+	notification_last_sent_date = frappe.db.get_default("vehicle_anniversary_notification_last_sent_date")
+	if notification_last_sent_date and getdate(notification_last_sent_date) == date_today:
+		return
+
+	vehicle_anniversary_data = get_vehicle_anniversary_data(date_today)
+
+	for d in vehicle_anniversary_data:
+		doc = frappe.get_doc("Vehicle Booking Order", d.name)
+		doc.send_customer_vehicle_anniversary_notification()
+
+	if vehicle_anniversary_data:
+		frappe.db.set_default("vehicle_anniversary_notification_last_sent_date", date_today)
```

Comment:
No need to check for condition .. if there's not vehicle anniversaries to send .. we should still update last sent date so that this function does not keep on running